src/pages/chainlink-functions/resources/add-functions-to-projects.md

```diff
@@ -0,0 +1,317 @@
+   ```
+
+### Configure Hardhat
+
+Modify `hardhat.config.js` to include the `0.8.7` and `0.8.9` compiler version and supported networks. See the [Supported Networks](/chainlink-functions/supported-networks) page for a list of networks where Chainlink Functions DONs are available. This example uses Polygon Mumbai by default:
+
+<CodeSample src="samples/Functions/adding-functions-to-projects/hardhat.config.js" />
+
+If you use this configuration with `require('dotenv').config()` to get your RPC URLs and private keys, you must also create a `.env` file with these variables set. Add your RPC URLs with your API keys and the private key to the wallet you will use to fund transactions on these testnets.
+
+```shell
+touch .env && open .env
+```
+
+```text
+# RPC URLs
+MUMBAI_RPC_URL="https://polygon-mumbai.infura.io/v3/your_api_key"
+SEPOLIA_RPC_URL="https://sepolia.infura.io/v3/your_api_key"
+
+# Wallet private key
+PRIVATE_KEY="your_private_key"
+```
+
+Run `npx hardhat compile` to make sure Hardhat is configured correctly. You will see several warnings, but everything is correct if the Solidity files compile successfully.
+
+## Get the dependency contracts and scripts
+
+Get the required libraries from the [Chainlink Functions Starter Kit](https://github.com/smartcontractkit/functions-hardhat-starter-kit). You can clone the repo and get the folders manually, or use a single command to get a tarball through the GitHub API and extract the folders you need to the correct location in your project:
+
+```shell
+curl -L -o ../functions.tar.gz https://api.github.com/repos/smartcontractkit/functions-hardhat-starter-kit/tarball/main &&
+tar -xf ../functions.tar.gz --strip=1 --wildcards smartcontractkit-functions-hardhat-starter-kit-*/contracts/dev smartcontractkit-functions-hardhat-starter-kit-*/contracts/test smartcontractkit-functions-hardhat-starter-kit-*/FunctionsSandboxLibrary smartcontractkit-functions-hardhat-starter-kit-*/contracts/FunctionsConsumer.sol
+```
+
+When you are done, you should have the necessary dependencies in the following directories:
+
+- `contracts/dev`
+- `contracts/test`
+- `FunctionsSandboxLibrary`
+
+Run `npx hardhat compile` again to make sure all the dependencies are satisfied. Add missing dependencies or Hardhat configuration options as necessary.
+
+## Configure on-chain resources
+
+The on-chain resources are critical for Chainlink Functions to process your requests.
+
+1. [Create a consumer contract](#create-a-consumer-contract)
+1. [Deploy a consumer contract](#deploy-a-consumer-contract)
+
+### Create a consumer contract
+
+Use the [FunctionsConsumer.sol](https://github.com/smartcontractkit/functions-hardhat-starter-kit/blob/main/contracts/FunctionsConsumer.sol), which is already in your `./contracts` folder. After you complete this tutorial, you can modify it to fit your needs and redeploy it.
+
+In general, a consumer contract requires several components:
+
+<ChainlinkFunctions section="functions-consumer" />
```

Comment:
i see this import in here:
`import "@chainlink/contracts/src/v0.8/dev/functions/FunctionsClient.sol";`

is this right? don't we have to manually reference the library in our project. My understanding is we don't have functions contracts in the NPM lib

Reply:
I corrected this in the import to say that the file must be downloaded manually and imported from within the project. See commit https://github.com/smartcontractkit/documentation/pull/1171/commits/619081a542894423bfb45b5dda8c18d7b628c7a0